data/mods/Magiclysm/enchantments/Sun_Mage.json

```
@@ -26,15 +26,9 @@
   {
     "type": "effect_on_condition",
     "id": "SUN_MAGE_BUFF",
-    "recurrence": [ "30 m", "60 m" ],
-    "condition": {
-      "and": [
-        { "compare_int": [ "hour", ">", { "const": 6 } ] },
-        { "compare_int": [ "hour", "<", { "const": 21 } ] },
-        { "u_has_trait": "SUN_MAGE" },
-        "u_is_outside"
-      ]
-    },
+    "recurrence": [ "5 m", "5 m" ],
+    "condition": { "and": [ "is_day", "u_is_outside", { "u_has_trait": "SUN_MAGE" } ] },
+    "deactivate_condition": { "not": { "or": [ "is_day", "u_is_outside", { "u_has_trait": "SUN_MAGE" } ] } },
```

```suggestion
    "deactivate_condition": { "not": { "u_has_trait": "SUN_MAGE" } },
```
You don't wanna have the is_day or u_is_outside here. The deactivate_condition disables the EoC until the player/npc gains/loses a trait or bionic. The sun comming up should not make the EoC start running again if it is deactivated. At least if the documentation is up to date.